docs/src/main/asciidoc/cdi-reference.adoc

```diff
@@ -1081,6 +1083,33 @@ NOTE: These endpoints are only available in the development mode, i.e. when you
 In the development mode, it is also possible to enable monitoring of business method invocations and fired events.
 Simply set the `quarkus.arc.dev-mode.monitoring-enabled` configuration property to `true` and explore the relevant Dev UI pages.
 
+[[strict_mode]]
+== Strict Mode
+
+By default, ArC does not perform all validations required by the CDI specification.
+It also improves CDI usability in many ways, some of them being directly against the specification.
+
+To be able to eventually pass the CDI Lite TCK, ArC also has a _strict_ mode.
+This mode enables additional validations and disables certain improvements that conflict with the specification.
+Applications that work as expected in the strict mode will function identically in the default, non-strict mode.
```

Comment:
Do you think we can make this promise long term?

Reply:
I believe so, yes. @mkouba WDYT about this statement? I think we should strive for this, actually.

Reply:
After discussion with @mkouba, I'll remove this sentence.

Reply:
Done.

Reply:
I'd rather remove this sentence. It's not entirely clear to me what does _"work as expected"_ mean and _"will function identically"_ would need some clarification as well. The _"This mode enables additional validations..."_ sentence is IMO good enough.